grafana-plugin/src/containers/IntegrationContainers/IntegrationTemplatesList.tsx

```diff
@@ -22,17 +22,33 @@ interface IntegrationTemplateListProps {
   templates: AlertTemplatesDTO[];
   alertReceiveChannelId: AlertReceiveChannel['id'];
   openEditTemplateModal: (templateName: string | string[]) => void;
+  alertReceiveChannelIsBasedOnAlertManager: boolean;
+  alertReceiveChannelAllowSourceBasedResolving: boolean;
 }
 
 const IntegrationTemplateList: React.FC<IntegrationTemplateListProps> = ({
   templates,
   openEditTemplateModal,
   alertReceiveChannelId,
+  alertReceiveChannelIsBasedOnAlertManager,
+  alertReceiveChannelAllowSourceBasedResolving,
 }) => {
   const { alertReceiveChannelStore } = useStore();
   const [isRestoringTemplate, setIsRestoringTemplate] = useState<boolean>(false);
   const [templateRestoreName, setTemplateRestoreName] = useState<string>(undefined);
   const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [autoresolveValue, setAutoresolveValue] = useState<boolean>(alertReceiveChannelAllowSourceBasedResolving);
+
+  const handleSaveClick = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setAutoresolveValue(event.target.checked);
+    alertReceiveChannelStore
+      .saveAlertReceiveChannel(alertReceiveChannelId, {
+        allow_source_based_resolving: event.target.checked,
+      })
+      .finally(() => {
+        openNotification('Autoresolve ' + (event.target.checked ? 'enabled' : 'disabled'));
```

Comment:
This might not be the correct approach as if the save request fails, it will hit finally clause letting the user know he autoresolved, but it didn't. This should go into a `then` instead

Reply:
thank you, fixed